components/engine/container/health_test.go

```diff
@@ -0,0 +1,15 @@
+package container
+
+import "testing"
+
+func TestOpenCloseMonitorChannel(t *testing.T) {
+	h := Health{}
+
+	for i := 0; i < 128; i++ {
+		go h.CloseMonitorChannel()
+		go h.CloseMonitorChannel()
+		go h.OpenMonitorChannel()
+		h.String()
```

Comment:
Looking at this code, I am curious if this will trip the race detector. We definitely set this at 56 then read it here. This isn't going to be a panic-level read, but it may get the wrong value.

Reply:
I ran with `-race` option as you suggested yesterday and got these results: